src/metabase/query_processor/middleware/add_dimension_projections.clj

```diff
@@ -166,47 +168,50 @@
     ;; fetch remapping column pairs if any exist...
     (if-let [infos (not-empty (remap-column-infos (concat fields breakout)))]
       ;; if they do, update `:fields`, `:order-by` and `:breakout` clauses accordingly and add to the query
       (let [ ;; make a map of field-id-clause -> fk-clause from the tuples
             original->remapped             (into {} (map (juxt :original-field-clause :new-field-clause)) infos)
             existing-fields                (add-fk-remaps-rewrite-existing-fields infos fields)
-            ;; don't add any new entries for fields that already exist. Use [[mbql.u/remove-namespaced-options]] here so we don't
-            ;; add new entries even if the existing Field has some extra info e.g. extra unknown namespaced keys.
+            ;; don't add any new entries for fields that already exist. Use [[mbql.u/remove-namespaced-options]] here so
+            ;; we don't add new entries even if the existing Field has some extra info e.g. extra unknown namespaced
+            ;; keys.
             existing-normalized-fields-set (into #{} (map mbql.u/remove-namespaced-options) existing-fields)
             new-fields                     (into
                                             existing-fields
                                             (comp (map :new-field-clause)
                                                   (remove (comp existing-normalized-fields-set mbql.u/remove-namespaced-options)))
                                             infos)
             new-breakout                   (add-fk-remaps-rewrite-breakout original->remapped breakout)
-            new-order-by                   (add-fk-remaps-rewrite-order-by original->remapped order-by)]
+            new-order-by                   (add-fk-remaps-rewrite-order-by original->remapped order-by)
+            remaps                         (into [] (comp cat (distinct)) [source-query-remaps (map :dimension infos)])]
         ;; return the Dimensions we are using and the query
-        {:remaps (into [] (comp cat (distinct)) [source-query-remaps (map :dimension infos)])
-         :query  (cond-> query
-                   (seq fields)   (assoc-in [:query :fields] new-fields)
-                   (seq order-by) (assoc-in [:query :order-by] new-order-by)
-                   (seq breakout) (assoc-in [:query :breakout] new-breakout))})
+        (cond-> query
+          (seq fields)   (assoc :fields new-fields)
+          (seq order-by) (assoc :order-by new-order-by)
+          (seq breakout) (assoc :breakout new-breakout)
+          (seq remaps)   (assoc ::remaps remaps)))
       ;; otherwise return query as-is
-      {:remaps source-query-remaps, :query query})))
+      (cond-> query
+        (seq source-query-remaps) (assoc ::remaps source-query-remaps)))))
+
+(s/defn ^:private add-fk-remaps :- QueryAndRemaps
+  "Add any Fields needed for `:external` remappings to the `:fields` clau se of the query, and update `:order-by`
```

Comment:
typo in `clau se`

Reply:
Same thing, I fixed this in #20009 and didn't carry that fix over to this PR when I merged `master` in to this one last night. I'll fix that too

Reply:
fixed